Use aws-sdk-s3 instead aws-sdk

tricknotes opened the pull request:
The aws-sdk gem version 3 is now released and it depends on the separated feature specific gems.

So this commit changes the dependent gem under the following document:
https://github.com/aws/aws-sdk-ruby/blob/master/V3_UPGRADING_GUIDE.md

eagletmt commented on s3_assets_uploader.gemspec:
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "aws-sdk", ">= 2.0"
+  spec.add_dependency "aws-sdk-s3", "~> 1.0"
```
Please do not use pessimistic version constraint. Specifying the upper limit rarely helps.
It should be `>= 1.0` .

tricknotes replied:
I'm OK.
I updated and rebased as https://github.com/eagletmt/s3_assets_uploader/pull/4/commits/396da6e04869475fb0173cb47982e1f61eb9f7b3 .

tricknotes committed:
Use aws-sdk-s3 instead aws-sdk

The aws-sdk gem version 3 is now released and it depends on the
separated feature specific gems.

So this commit changes the dependent gem under the following document:
https://github.com/aws/aws-sdk-ruby/blob/master/V3_UPGRADING_GUIDE.md

eagletmt commented:
Released v0.4.0 💎 
https://rubygems.org/gems/s3_assets_uploader/versions/0.4.0
